bindsnet/datasets/alov300.py

```diff
@@ -0,0 +1,318 @@
+        x = np.array(x)
+        y = np.array(y)
+        self.len = len(y)
+        print("ALOV dataset parsing done.")
+        print("Total number of annotations in ALOV dataset = %d" % num_anno)
+        return x, y
+
+    def get_sample(self, idx):
+        """
+        Returns sample without transformation for visualization.
+
+        Sample consists of resized previous and current frame with target
+        which is passed to the network. Bounding box values are normalized
+        between 0 and 1 with respect to the target frame and then scaled by
+        factor of 10.
+        """
+        opts_curr = {}
+        curr_sample = {}
+        curr_img = self.get_orig_sample(idx, 1)["image"]
+        currbb = self.get_orig_sample(idx, 1)["bb"]
+        prevbb = self.get_orig_sample(idx, 0)["bb"]
+        bbox_curr_shift = BoundingBox(prevbb[0], prevbb[1], prevbb[2], prevbb[3])
+        (
+            rand_search_region,
+            rand_search_location,
+            edge_spacing_x,
+            edge_spacing_y,
+        ) = cropPadImage(bbox_curr_shift, curr_img)
+        bbox_curr_gt = BoundingBox(currbb[0], currbb[1], currbb[2], currbb[3])
+        bbox_gt_recentered = BoundingBox(0, 0, 0, 0)
+        bbox_gt_recentered = bbox_curr_gt.recenter(
+            rand_search_location, edge_spacing_x, edge_spacing_y, bbox_gt_recentered
+        )
+        curr_sample["image"] = rand_search_region
+        curr_sample["bb"] = bbox_gt_recentered.get_bb_list()
+
+        # additional options for visualization
+        opts_curr["edge_spacing_x"] = edge_spacing_x
+        opts_curr["edge_spacing_y"] = edge_spacing_y
+        opts_curr["search_location"] = rand_search_location
+        opts_curr["search_region"] = rand_search_region
+
+        # build prev sample
+        prev_sample = self.get_orig_sample(idx, 0)
+        prev_sample, opts_prev = crop_sample(prev_sample)
+
+        # scale
+        scale = Rescale((self.input_size, self.input_size))
+        scaled_curr_obj = scale(curr_sample, opts_curr)
+        scaled_prev_obj = scale(prev_sample, opts_prev)
+        training_sample = {
+            "previmg": scaled_prev_obj["image"],
+            "currimg": scaled_curr_obj["image"],
+            "currbb": scaled_curr_obj["bb"],
+        }
+        return training_sample, opts_curr
+
+    def get_orig_sample(self, idx, i=1):
+        """
+        Returns original image with bounding box at a specific index.
+        Range of valid index: [0, self.len-1].
+        """
+        curr = cv2.imread(self.x[idx][i])
+        curr = bgr2rgb(curr)
+        currbb = self.get_bb(self.y[idx][i])
+        sample = {"image": curr, "bb": currbb}
+        return sample
+
+    def get_bb(self, ann):
+        """
+        Parses ALOV annotation and returns bounding box in the format:
+        [left, upper, width, height]
+        """
+        ann = ann.strip().split(" ")
+        left = min(float(ann[1]), float(ann[3]), float(ann[5]), float(ann[7]))
```

Comment:
Instead of calling `float` on all these tokens separately, is there a reason the `ann` initialization cannot be this?
```
# Renamed ann to coordinates because it's usually not best practice to re-assign an argument.
coordinates = map(lambda x: float(x), ann.strip().split(" "))
```

Reply:
That is a much better way to do this thanks!